### PR TITLE
feat: Add register function

### PR DESCRIPTION
**Pull request by jn2657**
# Description

Based on the previous [PR](https://github.com/SE-8-2021/frontend/pull/40), I add a simple register function to let users can register accounts by themselves.
In addition, to improve this function, there are some constraints for password security can be added.
Also, one of the console error is fixed in this PR. (Change `background` to `background="true"`)

# ScreenShot

截圖 2022-01-08 下午11 08 42
截圖 2022-01-08 下午11 08 59

### CODE REVIEW

File: src/App/component/Login.jsx [hunk 1]
```diff
@@ -32,7 +34,14 @@ export default function Login() {
   const history = useHistory()
   const [username, setUsername] = useState("")
   const [password, setPassword] = useState("")
-  const [invalidAccount, setInvalidAccount] = useState(false)
+  const [accountOperationHint, setAccountOperationHint] = useState("")
+  const [open, setOpen] = useState(false);
```
Comment (Xanonymous-GitHub) [inline src/App/component/Login.jsx]: Suggest clarifying who use these states.
```suggestion
  const [isWhatOpen, setWhatOpen] = useState(false);
```

File: src/App/component/Login.jsx [hunk 2]
```diff
@@ -32,7 +34,14 @@ export default function Login() {
...
+  const handleClose = () => {
```
Comment (Xanonymous-GitHub) [inline src/App/component/Login.jsx]:
```suggestion
  const handleWhatClose = () => {
```

File: src/App/component/Login.jsx [hunk 3]
```diff
@@ -32,7 +34,14 @@ export default function Login() {
...
+    setOpen(false);
+  };
+  const handleToggle = () => {
```
Comment (Xanonymous-GitHub) [inline src/App/component/Login.jsx]:
```suggestion
  const handleWhatToggle = () => {
```

File: src/App/component/Login.jsx [hunk 4]
```diff
@@ -42,18 +51,44 @@ export default function Login() {
...
+    } else if (!passwordRegex.test(password)) {
+      alert("Password should contains: \n 1. More than 8 digits\n 2. At least one uppercase and lowercase character\n 3. At least one number\n 4. At least one symbol")
+    } else {
+      let payload = {
+        username: username,
+        password: password
+      }
+      try {
+        const response = await Axios.post(`http://localhost:9100/pvs-api/auth/register`, payload)
+        response.data ? setAccountOperationHint("registerSuccess") : setAccountOperationHint("registerFailed")
```
Comment (Xanonymous-GitHub) [inline src/App/component/Login.jsx:83]:
```suggestion
        response.data ? 
        setAccountOperationHint("Account registered successfully.") : 
        setAccountOperationHint("Account registration failed!")
```

File: src/App/component/Login.jsx (hunk 1 above)
Comment (VincentETN) [reply src/App/component/Login.jsx]: 不是改成 What 的吧？ XD
應該會是 backdropOpen or circularProgressOpen 之類的。

File: src/App/component/Login.jsx [hunk 5]
```diff
@@ -105,16 +149,22 @@ export default function Login() {
...
-        {/* <button onClick={login} >Login</button> */}
-        <Button variant="contained" onClick={login} color="primary">
-          Login
-        </Button>
+        <span>
+          {/* <button onClick={login} >Login</button> */}
+          <Button variant="contained" onClick={register} color="primary">
+            Register
+          </Button>
+          {' '}
```
Comment (VincentETN) [inline src/App/component/Login.jsx]: 這間隔也許可以用 styles 處理。

**Comment by hellenGUAN** [comment]
小小建議
紅字可以改成：
Account already **exists**
Account **is registered** successfully

File: src/App/component/Login.jsx (hunk 1 above)
Comment (Xanonymous-GitHub) [reply src/App/component/Login.jsx]: Of course my friend, it is just an example.

Comment (Xanonymous-GitHub) [reply src/App/component/Login.jsx]: OMG, he did it ....
@jn2657 Please don't use `What` for this naming, it is just an example, the real name should not be `What` ...

**Review by hellenGUAN — APPROVED** [review]

**Review by VincentETN — APPROVED** [review]

**Review by Xanonymous-GitHub — APPROVED** [review]
LGTM

**Pull request merged**